addons/mail/static/src/core/common/suggestion_model.js

```diff
@@ -0,0 +1,58 @@
+import { OR, Record } from "@mail/core/common/record";
+
+export class Suggestion extends Record {
+    static id = OR("mention", "channelCommand", "cannedResponse");
+    /** @type {Object.<number, import("models").Suggestion>} */
+    static records = {};
+    /** @returns {import("models").Suggestion} */
+    static get(data) {
+        return super.get(data);
+    }
+    /** @returns {import("models").Suggestion|import("models").Suggestion[]} */
+    static insert(data) {
+        return super.insert(...arguments);
+    }
+    static new() {
+        return super.new(...arguments);
+    }
+
+    mention = Record.one("Mention");
+    channelCommand;
+    cannedResponse = Record.one("CannedResponse");
+
+    get type() {
+        if (this.mention) {
+            return "mention";
+        }
+        if (this.channelCommand) {
+            return "channelCommand";
+        }
+        if (this.cannedResponse) {
+            return "cannedResponse";
+        }
+        throw new Error("Unkown suggestion type");
```

Comment:
be careful to throw an Error, it blocks the workflow. I think this condition is not serious enough to block the workflow. Just return undefined or `""` is enough.